x/ccv/provider/types/proposal.go

```diff
@@ -135,6 +137,12 @@ func (cccp *ConsumerAdditionProposal) ValidateBasic() error {
 		return errorsmod.Wrap(ErrInvalidConsumerAdditionProposal, "unbonding period cannot be zero")
 	}
 
+	// Top N corresponds to the top N% of validators that have to validate the consumer chain and can only be 0 (for an
+	// Opt In chain) or in the range [50, 100] (for a Top N chain).
+	if cccp.Top_N != 0 && cccp.Top_N < 50 || cccp.Top_N > 100 {
```

Comment:
Should we allow more generous values for Top_N here?

Reply:
Rationale behind 50 being a lower bound is described [here](https://github.com/cosmos/interchain-security/blob/main/docs/docs/adrs/adr-015-partial-set-security.md):
> In case of a Top N chain, we restrict the possible values of top_N from (0, 100] to [50, 100]. By having top_N >= 50 we can guarantee that we cannot have a successful attack, assuming that at most 1/3 of provider validators can be malicious. This is because, a Top N chain with N >= 50% would have at least 1/3 honest validators, which is sufficient to stop attacks. Additionally, by having N >= 50% (and hence N > (VetoThreshold = 33.4%)) we enable the top N validators to Veto any ConsumerAdditionProposal for consumer chains they do not want to validate.

Additionally, with `top_N >= 50` we also do **not** need to have fraud votes for _Top N_ chains. This is because, if we assume < 1/3 of validators can be malicious on the Hub, then a consumer with 50% of the Hub's power would have >= 1/3 of honest validators. In order to perform incorrect execution we need >2/3 of malicious validators and hence with `top_N >= 50` we cannot have incorrect execution and therefore do not need fraud votes.
Not having fraud votes for _Top N_ chains is also nice because validators in a _Top N_ chain might have been forced to validate and hence we want to reduce the potentiality of them getting slashed for something they were forced to do.

Reply:
nit: add parentheses for clarity